### PR TITLE
switched to new config object format

### PR DESCRIPTION
**Pull request by pelger**

### CODE REVIEW

File: runner.js [hunk 1]
```diff
@@ -16,11 +16,11 @@
...
 
-module.exports = function (config) {
-  var puller = require('./lib/gitPuller')(config)
-  var builder = require('./lib/processBuilder')(config)
-  var previewer = require('./lib/previewer')(config)
-  var system = require('./lib/system')(config)
+module.exports = function () {
+  var puller = require('./lib/gitPuller')()
+  var builder = require('./lib/processBuilder')()
+  var previewer = require('./lib/previewer')()
```
Comment (davidmarkclements) [inline runner.js]: inline requiring is bad practice

File: runner.js [hunk 2]
```diff
@@ -49,8 +49,8 @@ module.exports = function (config) {
   var preview = function (sysDef, name, cb) { previewer.preview(sysDef, name, cb) }
   var previewAll = function (sysDef, cb) { previewer.previewAll(sysDef, cb) }
 
-  var grep = function (name, config, search, cb) { util.grep(name, config, search, cb) }
-  var grepAll = function (system, config, search, cb) { util.grepAll(system, config, search, cb) }
+  var grep = function (name, search, cb) { util.grep(name, search, cb) }
+  var grepAll = function (system, search, cb) { util.grepAll(system, search, cb) }
```
Comment (davidmarkclements) [inline runner.js]: You could just do `var grep = util.grep`, same for grepAll

File: test/directoryWatcherTest.js [hunk 3]
```diff
@@ -20,33 +20,21 @@ var test = require('tape')
...
-      'frontend_PORT': 10000,
-      'users_PORT': 10001,
-      'permissions_PORT': 10002,
-      'business_logic_PORT': 10003,
-      'audit_PORT': 10004,
-      'emails_PORT': 10005,
-      SERVICE_HOST: '0.0.0.0',
-      SERVICE_PORT: 20008
-    }
+  path: path.join(__dirname, '/fixture/directoryWatcher'),
```
Comment (davidmarkclements) [inline test/directoryWatcherTest.js]: Should be joining instead of using / , replace / wtih ','

File: test/processRunnerTest.js [hunk 4]
```diff
@@ -18,50 +18,36 @@ var test = require('tape')
...
-      'users_PORT': 10001,
-      'permissions_PORT': 10002,
-      'business_logic_PORT': 10003,
-      'audit_PORT': 10004,
-      'emails_PORT': 10005,
-      SERVICE_HOST: '0.0.0.0',
-      SERVICE_PORT: 20008
-    }
+  type: 'node',
+  path: path.join(__dirname, '/fixture/directoryWatcher'),
```
Comment (davidmarkclements) [inline test/processRunnerTest.js]: Same, shouldn't use /

File: lib/processRunner.js [hunk 5]
```diff
@@ -75,46 +46,24 @@ module.exports = function () {
...
-        options.windowsVerbatimArguments = true
 
-        // /s encodes inner quotes so the whole command itself can be quoted correctly
-        child = spawn(process.env.comspec, ['/s', '/c', '"', toExec, '"'], options)
+        // allows quoted args to cmd.exe, /s encodes inner quotes so the whole command itself can be quoted correctly
+        options.windowsVerbatimArguments = true
+        child = spawn(process.env.comspec, ['/s', '/c', '"', cmd, '"'], options)
       } else {
-        child = spawn('/bin/bash', ['-c', toExec], options)
+        child = spawn('/bin/bash', ['-c', cmd], options)
```
Comment (davidmarkclements) [inline lib/processRunner.js]: From discussions, I thought we we're going to spawn node directly (or whichever process) instead of using bash/com?

File: lib/processRunner.js [hunk 6]
```diff
@@ -75,46 +46,24 @@ module.exports = function () {
...
-    }
-
-    if (isWin) {
-      toExec = envArgs + ' ' + cmd
-    } else {
-      toExec = envArgs + 'bash -c "exec ' + cmd + '"'
+    env = Object.create(process.env)
+    if (container.environment) {
+      env = _.merge(env, container.environment)
     }
```
Comment (davidmarkclements) [inline lib/processRunner.js]: process.env is key value only, no need for depth merging, no need to lodash if we're 4+ we can just use Object.assign

File: lib/processRunner.js [hunk 7]
```diff
@@ -20,52 +20,23 @@ var spawn = require('child_process').spawn
...
     var env
-    var toExec
     var child = {}
-    var envArgs = ''
     var called = false
 
-    if (cmd.slice(0, 5) === 'node ' && cmd.slice(0, 7) !== 'node -r') {
-      // use same node running fuge
-      // and quote windows path to handle potential spaces
+    console.log('running: ' + container.run)
```
Comment (mcollina) [inline lib/processRunner.js]: is this spurious?

File: lib/processRunner.js [hunk 8]
```diff
@@ -75,46 +46,24 @@ module.exports = function () {
       container.type = 'node'
     }
```
Comment (mcollina) [inline lib/processRunner.js]: we should probably remove all of this profiling stuff.

File: lib/processRunner.js (hunk 6 above)
Comment (mcollina) [inline lib/processRunner.js]: Definitely, no `_` needed.

File: lib/processRunner.js (hunk 5 above)
Comment (mcollina) [inline lib/processRunner.js]: Yes. We should not spawn bash, but use `node` directly.

Comment (pelger) [inline lib/processRunner.js]: Yes - that will be subject of an upcoming separate PR :)

File: lib/processRunner.js (hunk 7 above)
Comment (pelger) [inline lib/processRunner.js]: no - expected in the fuge output at the moment - although I agree should probably be moved to the main fuge module

File: runner.js (hunk 1 above)
Comment (pelger) [inline runner.js]: Done

File: runner.js (hunk 2 above)
Comment (pelger) [inline runner.js]: yep - pending PR from @calebissharp that cleans up this file - will merge once this is in!

File: test/directoryWatcherTest.js (hunk 3 above)
Comment (pelger) [inline test/directoryWatcherTest.js]: Done

File: test/processRunnerTest.js (hunk 4 above)
Comment (pelger) [inline test/processRunnerTest.js]: done

File: lib/processRunner.js (hunk 6 above)
Comment (pelger) [inline lib/processRunner.js]: Fair point and it's used in a few places where it's not really needed - lets leave for now and I'll review and send some separate PRs removing lodash from the whole codebase
